android/app/src/main/java/org/electroncash/electroncash3/NewWallet.kt

```diff
@@ -113,16 +199,79 @@ abstract class NewWalletDialog2 : TaskLauncherDialog<String>() {
+
+        val multisig = arguments!!.getBoolean("multisig")
+
+        /**
+         * For multisig wallets, wait until all cosigners have been added,
+         * and then create and load the multisig wallet.
+         *
+         * Otherwise, load the created wallet.
+         */
+        if (multisig) {
+            val currentCosigner = arguments!!.getInt("i_signer")
+            val numCosigners = arguments!!.getInt("cosigners")
+            val numSignatures = arguments!!.getInt("signatures")
+
+            // Close the previous cosigner's keystore dialog.
+            // TODO: Consider a better solution for closing dialogs?
+            (targetFragment as KeystoreDialog).dismiss()
+
+            if (currentCosigner == numCosigners) {
+                daemonModel.commands.callAttr(
+                        "create_multisig", name, password,
+                        Kwarg("keystores", keystores.toArray()),
+                        Kwarg("cosigners", numCosigners),
+                        Kwarg("signatures", numSignatures)
+                )
+                daemonModel.loadWallet(name, password)
+                closeDialogs(targetFragment!!)
+            }
+        } else {
+            daemonModel.loadWallet(name, password)
+        }
+
         return name
     }
 
     abstract fun onCreateWallet(name: String, password: String)
 
     override fun onPostExecute(result: String) {
-        (targetFragment as NewWalletDialog1).dismiss()
-        daemonModel.commands.callAttr("select_wallet", result)
-        (activity as MainActivity).updateDrawer()
+
+        val multisig = arguments!!.getBoolean("multisig")
+
+        /**
+         * For multisig wallets, we need to first show the master key to the 1st cosigner, and
+         * then prompt for data for all other cosigners by calling the KeystoreDialog again.
+         */
+        if (multisig) {
+
+            val currentCosigner = arguments!!.getInt("i_signer")
+            val numCosigners = arguments!!.getInt("cosigners")
+
+            if (currentCosigner < numCosigners) {
+                // The first cosigner sees their master public key; others are prompted for data
+                val nextDialog: DialogFragment = if (currentCosigner == 1) {
+                    MasterPublicKeyDialog()
+                } else {
+                    KeystoreDialog()
+                }
+
+                arguments!!.putInt("i_signer", currentCosigner + 1)
+
+                nextDialog.setArguments(arguments)
+                showDialog(this, nextDialog)
+            } else { // last cosigner done; finalize wallet
+                daemonModel.commands.callAttr("select_wallet", result)
+                (activity as MainActivity).updateDrawer()
+            }
+        } else {
+            // In a standard wallet, close the dialogs and open the newly created wallet.
+            (targetFragment as KeystoreDialog).dismiss()
+            (targetFragment!!.targetFragment as NewWalletDialog1).dismiss()
```

Comment:
Since you've added a recursive closeDialogs function, you might as well use it here.